ftdc/ftdc.go

```diff
@@ -355,23 +369,85 @@ func (ftdc *FTDC) writeDatum(datum datum) error {
 // getWriter returns an io.Writer xor error for writing schema/data information. `getWriter` is only
 // expected to be called by `newDatum`.
 func (ftdc *FTDC) getWriter() (io.Writer, error) {
-	if ftdc.outputWriter != nil {
+	// If we have an `outputWriter` without a `currOutputFile`, it means ftdc was constructed with
+	// an explicit writer. We will use the passed in writer for all operations.
+	if ftdc.outputWriter != nil && ftdc.currOutputFile == nil {
```

Comment:
This `getWriter` call is where all the new magic happens. Please judge the documentation. The goal is to make the relationship between the `outputWriter` and `currOutputFile` clear for both the `NewWithWriter` and `New[WithoutWriter]` FTDC constructors.

But I also suspect what I'm doing there is a bit of an advanced technique and not easy to immediately grok.